pkg/odo/genericclioptions/runnable.go

```diff
@@ -248,16 +256,23 @@ func GenericRun(o Runnable, testClientset clientset.Clientset, cmd *cobra.Comman
 			}
 		}
 
-		var devfilePath, componentName string
-		var devfileObj *parser.DevfileObj
-		devfilePath, devfileObj, componentName, err = getDevfileInfo(cmd, deps.FS, cwd, variables, userConfig.GetImageRegistry())
-		if err != nil {
-			startTelemetry(cmd, err, startTime)
-			return err
+		useDevfile := true
+		if devfileUser, ok := o.(DevfileUser); ok {
+			useDevfile = devfileUser.UseDevfile(ctx, cmdLineObj, args)
+		}
+
+		if useDevfile {
```

Comment:
I think we no longer need the `devfile-not-needed` annotation added to some Cobra commands. I remember it was introduced in #6905 to fix a similar issue when running `odo analyze`. Can you remove this as well?
- https://github.com/feloy/odo/blob/bugfix-6971/odo-registry-invalid-devfile/pkg/odo/genericclioptions/devfile.go#L22
- https://github.com/feloy/odo/blob/bugfix-6971/odo-registry-invalid-devfile/pkg/odo/genericclioptions/devfile.go#L37